### ISSUE TITLE
IP bans prevent login which precludes running banstick commands on the account

### ISSUE THREAD

**Issue by wingzero54** [CivClassic/BanStick#28]
If an account has never logged in, you cannot run banstick commands on it. And IP bans prevent an account from logging in, meaning there's no way to forgive them from the ip ban. This provides more legwork for us, as we have to investigate much deeper on the ip ban, instead of just working with the player's account to allow it through. 

Need to allow the player to connect and then ban them

**Comment by ProgrammerDan** [comment]
https://github.com/CivClassic/BanStick/blob/master/src/main/java/com/programmerdan/minecraft/banstick/handler/BanStickEventHandler.java#L130

IP Ban occurs after UUID/playername registration occurs. 

This should already work as you're describing. The only thing you cannot do at present is "pre-authorize" a specific UUID or playername without some DB muckwork; however from an admin perspective have them attempt to login once and you'll be able to manage their bans immediately.

If that's not the case (which according to the code is the case) perhaps there is something you're running on a Bungee node that kicks them before the code I pasted, or there is a bug and this isn't an enhancement.

**Comment by wingzero54** [comment]
The example I had earlier was we had a /18 ip range ban. A player was trying to connect from within that range, and was being refused a connection. So I could not /forgive player ip to allow them to bypass the range ban, because the uuid was not registered with banstick. Instead, I had to forgive the ip range ban to let the player login, and would then need to /forgive player ip and then /banstick 192.168.10.10/18 to reban the IP-range. 

As you say, I'm not sure why - may be a bug, or some other interaction.

**Comment by wingzero54** [comment]
It only specifically occurs with ip range bans, so there may be some separate mechanic going on with ip range bans

**Comment by ProgrammerDan** [comment]
I wonder if the pre-reg is what isn't working; that's definitely possible. The IP range check occurs even further on beyond the IP check, so that's curious.

**Comment by wingzero54** [comment]
I've noticed this also happens with transitive bans. "Preventing login by ... due to Player Ban .... transitively applied from ..." Attempted logins like this are not registered with a session, meaning you are not associated with the account you tried to login with. Although I haven't looked into it, and not sure if it is meant to register a session when login is prevented in transitive ban cases

**Comment by ProgrammerDan** [comment]
Ah, that would make more sense. A session isn't created until login
succeeds. We could move that process forward, however, if the lack of
session is what's preventing interaction with the user record....

**Comment by wingzero54** [comment]
Ran into this issue again, a player is being prevented from login due to an IP range ban, and he has no sessions so I cannot forgive him. And I cannot seem to forgive the IP range ban either, as it is saying it can't find the IP. I wonder if I could find out his IP and forgive specifically his IP, instead of his player

**Comment by Convoy20** [comment]
Did you use his UUID instead of username?

**Comment by wingzero54** [comment]
Oh dang that worked, I guess without a first session it didn't do the civ naming thing

**Comment by Convoy20** [comment]
yeah we have that issue on devhell lobby all the time, dunno why and it's a pain in the butt

**Comment by ProgrammerDan** [comment]
I didn't include an external lookup if the uuid/name pair is unknown. And
apparently some things aren't executing in my intended order, as the
session record that would create a banstick local pairing, does not appear
to be created in case of range or vps ban. PITA
